### ISSUE TITLE
[Feature request] gcs_fuse_csi_driver_config to enable the fuse csi driver on the cluster

### ISSUE THREAD

**Issue by Billyzou0741326** [Neutrollized/free-tier-gke#11]
I recently stumbled upon this [documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/cloud-storage-fuse-csi-driver#enable) on enabling the cloud storage fuse csi driver. According to the docs, doing so allows mounting gcs buckets as ephemeral or persistent volume - something I'd like to experiment with.

Do you find it plausible to introduce a terraform variable to enable the csi driver? Or would this be something that's better kept in my fork?

What I have in mind is something like:

```diff
--- a/variables.tf
+++ b/variables.tf
@@ -222,6 +222,12 @@ variable "filestore_csi_driver_enabled" {
   default     = false
 }

+variable "gcs_fuse_csi_driver_enabled" {
+  description = "When enabled, allows gcs buckets to be mounted as ephemeral or persistent volumes using the FUSE csi driver"
+  type        = bool
+  default     = false
+}

 variable "horizontal_pod_autoscaling_disabled" {
   description = "When enabled, allows increase/decrease number of replica pods based on resource usage of existing pods."
   type        = bool
```

and then

```diff
--- a/main.tf
+++ b/main.tf
@@ -157,6 +157,10 @@ resource "google_container_cluster" "primary" {
     config_connector_config {
       enabled = var.config_connector_enabled
     }
+
+    gcs_fuse_csi_driver_config {
+      enabled = var.gcs_fuse_csi_driver_enabled
+    }
   }
 }
```

**Comment by Neutrollized** [comment]
Yeah, I absolutely can!  Give me 'til the end of the week (pretty busy atm )

**Comment by Neutrollized** [comment]
HI @Billyzou0741326 ,

Thanks for making the request.  I've released some updates that includes the GCSFuse toggle you asked for.  I also took this opportunity to consolidate and organize some settings and variables.  If you want to toggle on GCSFuse, you will need to set:

```
addons_config = {
  gcs_fuse_csi_driver_enabled = true
}
```

See the [CHANGELOG](https://github.com/Neutrollized/free-tier-gke/blob/master/CHANGELOG.md#0180---2024-09-16) for more details.

Cheers,
GY
